src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/PackageItemViewModel.cs

```diff
@@ -839,6 +911,36 @@ private async Task ReloadPackageMetadataAsync()
             }
         }
 
+        private async Task UpdatePackageMaxVulnerabilityAsync(PackageIdentity packageIdentity, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            IEnumerable<PackageVulnerabilityMetadataContextInfo> vulnerabilityInfoList =
+                        await _vulnerabilityService.GetVulnerabilityInfoAsync(packageIdentity, cancellationToken);
+
+            SetVulnerabilityMaxSeverity(packageIdentity.Version, vulnerabilityInfoList?.FirstOrDefault()?.Severity ?? -1);
```

Comment:
If you're getting a list of vulnerabilities, would it make sense to select the one with the highest severity, instead of just the first, for setting the max severity?

Reply:
So, I wasn't able to locate where we explain this, but my understanding is that they are already sorted by severity when we query them from the database so there is no need to "sort" again.

[This code was just moved into another function](https://github.com/NuGet/NuGet.Client/pull/6044/files/85f658abf961201b1ba8dea5509aa913e61b2774#diff-118d6d1c3dcb2549a9436b8dafadcf915e0ae147a2ed54f3d82767252ee94701L817), but the implementation is the same and is being used in project PM UI too.